test/Identity.Test/IdentityServer/BaseRequestValidatorTests.cs

```diff
@@ -276,6 +281,99 @@ public async Task ValidateAsync_GrantTypes_OrgSsoRequiredTrue_ShouldSetSsoResult
         Assert.Equal("SSO authentication is required.", errorResponse.Message);
     }
 
+    // Test grantTypes with RequireSsoPolicyRequirement when feature flag is enabled
+    [Theory]
+    [BitAutoData("password")]
+    [BitAutoData("webauthn")]
+    [BitAutoData("refresh_token")]
+    public async Task ValidateAsync_GrantTypes_WithPolicyRequirementsEnabled_OrgSsoRequiredTrue_ShouldSetSsoResult(
+        string grantType,
+        [AuthFixtures.ValidatedTokenRequest] ValidatedTokenRequest tokenRequest,
+        CustomValidatorRequestContext requestContext,
+        GrantValidationResult grantResult)
+    {
+        // Arrange
+        _featureService.IsEnabled(FeatureFlagKeys.PolicyRequirements).Returns(true);
+        var context = CreateContext(tokenRequest, requestContext, grantResult);
+        context.CustomValidatorRequestContext.CaptchaResponse.IsBot = false;
+        _sut.isValid = true;
+
+        context.ValidatedTokenRequest.GrantType = grantType;
+        // Configure requirement to require SSO and exempt accepted users
+        var requirement = new RequireSsoPolicyRequirement(
+            [
+                new PolicyDetails
+                {
+                    PolicyType = PolicyType.RequireSso,
+                    OrganizationUserStatus = OrganizationUserStatusType.Confirmed
+                }
+            ]);
+        _policyRequirementQuery.GetAsync<RequireSsoPolicyRequirement>(Arg.Any<Guid>()).Returns(requirement);
```

Comment:
This is more test setup than we want when we're not testing the requirement itself. For example, this validator class doesn't care about the `OrganizationUserStatus`, t just wants to make sure that it throws an error if `RequireSso == true`.

I suggest that the policy filtering logic is moved to the constructor of the `RequireSsoPolicyRequirement`, which sets literal values for `CanUsePasskeyLogin` and `RequireSso` rather than using getter logic.  Then make those properties init-only. That way we still get the same behavior, but in these tests you can just do `var requirement = new RequireSsoPolicyRequirement { RequireSso = true };` which is a much nicer interface.

Reply:
Exellent suggestion -- Its way simpler!